packages/context/src/binding.ts

```diff
@@ -263,24 +263,31 @@ export class Binding<T = BoundValue> extends EventEmitter {
   private _cache: WeakMap<Context, T>;
   private _getValue?: ValueFactory<T>;
 
-  private _valueConstructor?: Constructor<T>;
-  private _providerConstructor?: Constructor<Provider<T>>;
-  private _alias?: BindingAddress<T>;
+  /**
+   * The original source value received from `to`, `toClass`, `toDynamicValue`,
+   * `toProvider`, or `toAlias`.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _source: any;
```

Comment:
Can we please avoid using `any`? At minimum, let's try to use `unknown` instead.

Alternatively, can we use a union type here?

```suggestion
  private _source: 
    | T // .to()
    | Constructor<T> // .toClass()
    | Constructor<Provider<T>> // .toProvider()
    | ValueFactory<T> | DynamicValueProviderClass<T> // .toDynamicValue()
    | BindingAddress<T> // .toAlias()
    ;
```

Reply:
I think it would be best to create a new type `SourceType<T>` and provide proper TSDoc comment to explain all different kinds of values the source can hold.

Reply:
I added `BindingSource` type per https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions.